### PR TITLE
Maryia/Allowing currency change if client has no real dmt5/dxtrade accounts

### PR DESCRIPTION
**Pull request by maryia-deriv**

### CODE REVIEW

File: src/javascript/app/pages/cashier/cashier.js [hunk 1]
```diff
@@ -84,43 +84,46 @@ const Cashier = (() => {
...
+            : localize('Your fiat account\'s currency is set to [_1].', `${currency}`);
+        const currency_message          = Currency.isCryptocurrency(currency)
             ? localize('This is your [_1] account.', `${Currency.getCurrencyDisplayCode(currency)}`)
-            : has_no_mt5 && has_no_transaction
-                ? localize('Your fiat account\'s currency is currently set to [_1].', `${currency}`)
-                : localize('Your fiat account\'s currency is set to [_1].', `${currency}`);
+            : currency_is_set_message;
 
-        const currency_hint = Currency.isCryptocurrency(currency)
+        const fiat_currency_change_hint = has_no_real_mt5 && has_no_real_dxtrade && has_no_transaction
```
Comment (maryia-deriv) [inline src/javascript/app/pages/cashier/cashier.js:121]: I introduced `currency_is_set_message` and `fiat_currency_change_hint` variables in order to eliminate eslint warning 'Do not nest ternary expressions'

File: src/javascript/app/pages/user/metatrader/metatrader.js [hunk 2]
```diff
@@ -210,7 +211,13 @@ const MetaTrader = (() => {
 
     const getAllAccountsInfo = (response) => {
         MetaTraderUI.init(submit, sendTopupDemo);
-        show_new_account_popup = Client.canChangeCurrency(State.getResponse('statement'), (response.mt5_login_list || []), false);
+        show_new_account_popup = Client.canChangeCurrency(
+            State.getResponse('statement'),
+            (response.mt5_login_list || []),
+            State.getResponse('trading_platform_accounts'),
+            Client.get('loginid'),
```
Comment (maryia-deriv) [inline src/javascript/app/pages/user/metatrader/metatrader.js:218]: passing `loginid` to `canChangeCurrency()` everywhere it's used according to the change of `canChangeCurrency`'s interface as part of https://github.com/binary-com/binary-static/pull/6814/files

File: src/javascript/app/pages/user/metatrader/metatrader.js [hunk 3]
```diff
@@ -19,7 +19,8 @@ const MetaTrader = (() => {
 
     const onLoad = () => {
         BinarySocket.send({ statement: 1, limit: 1 });
-        BinarySocket.wait('landing_company', 'get_account_status', 'statement').then(async () => {
+        BinarySocket.send({ trading_platform_accounts: 1, platform: 'dxtrade' });
+        BinarySocket.wait('landing_company', 'get_account_status', 'statement', 'trading_platform_accounts').then(async () => {
```
Comment (maryia-deriv) [inline src/javascript/app/pages/user/metatrader/metatrader.js:23]: Having requested a Deriv X accounts list, I get it from `trading_platform_accounts`.

File: src/javascript/app/pages/cashier/cashier.js [hunk 4]
```diff
@@ -84,43 +84,42 @@ const Cashier = (() => {
...
+        const can_change                 = Client.canChangeCurrency(statement, mt5_logins, dxtrade_accounts_list, Client.get('loginid'));
+        const has_upgrade                = upgrade_info.can_upgrade || upgrade_info.can_open_multi || can_change;
+        const account_action_text        = has_upgrade ? `<br />${localize('[_1]Manage your accounts[_2]', [`<a href=${Url.urlFor('user/accounts')}>`, '</a>'])}` : '';
+        const is_iom_client              = Client.get('residence') === 'im' || State.getResponse('website_status.clients_country') === 'im';
+        const change_text_for_iom        = is_iom_client ? localize('time') : localize('time or create a real MT5 account (or a real Deriv X account at deriv.com)');
+        const url_user_account           = `<a href=${Url.urlFor('user/accounts')}>`;
+
+        const criteria_not_met_message   = has_no_transaction
+            ? localize('You can no longer change the currency because you\'ve created a real MT5 account (or a real Deriv X account at deriv.com).') + account_action_text
+            : localize('You can no longer change the currency because you\'ve made a first-time deposit.') + account_action_text;
```
Comment (maryia-deriv) [inline src/javascript/app/pages/cashier/cashier.js:105]: Replaced an unnecessary `missingCriteria()` function with a `criteria_not_met_message` variable.

File: src/javascript/app/pages/cashier/cashier.js [hunk 5]
```diff
@@ -84,43 +84,42 @@ const Cashier = (() => {
...
-
-        const missingCriteria = (has_mt5, has_transaction) => {
-            const existing_mt5_msg          = localize('You can no longer change the currency because you\'ve created an MT5 account.') + account_action_text;
-            const existing_transaction_msg  = localize('You can no longer change the currency because you\'ve made a first-time deposit.') + account_action_text;
-
-            return has_mt5 && !has_transaction ? existing_mt5_msg : existing_transaction_msg;
-        };
+    const showCurrentCurrency = (currency, statement, mt5_logins, dxtrade_accounts_list) => {
+        const has_no_real_mt5_or_dxtrade = (!mt5_logins || mt5_logins.filter((account) => account.account_type !== 'demo').length === 0) &&
+            (!dxtrade_accounts_list || dxtrade_accounts_list.filter(account => account.account_type !== 'demo').length === 0);
```
Comment (matin-binary) [inline src/javascript/app/pages/cashier/cashier.js:89]: Suggestion: You can created a shared function out of this and reuse it in `client-base.js` as well.

**Review by mahdiyeh-fs — DISMISSED** [review]

File: src/javascript/app/pages/user/metatrader/metatrader.js [hunk 6]
```diff
@@ -262,8 +269,10 @@ const MetaTrader = (() => {
 
     const submit = (e) => {
         e.preventDefault();
+        const $btn_submit = $(e.target);
+        const acc_type    = $btn_submit.attr('acc_type');
 
-        if (show_new_account_popup) {
+        if (acc_type.includes('real') && show_new_account_popup) {
```
Comment (maryia-deriv) [inline src/javascript/app/pages/user/metatrader/metatrader.js:275]: showing a confirmation popup to a user, who is still able to change currency, only upon a real (not demo) account creation

File: src/javascript/app/pages/cashier/cashier.js (hunk 5 above)
Comment (maryia-deriv) [reply src/javascript/app/pages/cashier/cashier.js]: done ✅  )

**Review by mahdiyeh-fs — APPROVED** [review]

**Review by matin-binary — APPROVED** [review]
Nicely done!

**Pull request merged**